### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by ordabayevy**
- Reformat docstrings to have argument description start on a new line
- Add args to the docstring of `DistributedAnnDataCollectionDataModule`
- Add missing `obs_columns` to the init of `DistributedAnnDataCollectionDataModule`
- Display documented `:special-members:` like `__iter__` in the docs

### CODE REVIEW

File: cellarium/ml/train/training_plan.py [hunk 1]
```diff
@@ -13,16 +13,22 @@
...
-    Lightning module task to train cellarium-ml modules.
+    Lightning module task to train ``cellarium.ml`` modules.
 
     Args:
-        module: A cellarium.ml module to train.
-        optim_fn: A Pytorch optimizer class, e.g., :class:`~torch.optim.Adam`. If ``None``,
+        module:
+            A ``cellarium.ml`` module to train.
+        optim_fn:
+            A Pytorch optimizer class, e.g., :class:`~torch.optim.Adam`. If ``None``,
```
Comment (fedorgrab) [inline cellarium/ml/train/training_plan.py:22]: Out of curiosity, what does Tilda do in this context?

**Review by fedorgrab — CHANGES_REQUESTED** [review]
Most of my comments pertain to cosmetics and such, so feel free to resolve the discussions if you think otherwise.

File: cellarium/ml/train/training_plan.py (hunk 1 above)
Comment (ordabayevy) [reply cellarium/ml/train/training_plan.py:22]: Tilde means to hide prefix in the class name and only display `Adam` in the docs.

**Comment by ordabayevy** [comment]
Thanks for the feedback @fedorgrab ! I addressed your comments and added further changes:
- Use `sphinx-autodoc-typehints` to add type hints and defaults to the doc. Removed all manually added default values.
- Added `sphinx-copybutton` extension
- Updated html links to be displayed properly
- Updated examples docstrings to display all command line args on a new line

**Review by fedorgrab — APPROVED** [review]

**Pull request merged**